src/TreeSupport.cpp

```diff
@@ -2246,42 +2504,46 @@ void TreeSupport::generateBranchAreas(std::vector<std::pair<LayerIndex, TreeSupp
 void TreeSupport::smoothBranchAreas(std::vector<std::unordered_map<TreeSupportElement*, Polygons>>& layer_tree_polygons)
 {
     double progress_total = TREE_PROGRESS_PRECALC_AVO + TREE_PROGRESS_PRECALC_COLL + TREE_PROGRESS_GENERATE_NODES + TREE_PROGRESS_AREA_CALC + TREE_PROGRESS_GENERATE_BRANCH_AREAS;
-    const coord_t max_radius_change_per_layer = 1 + config.support_line_width / 2; // this is the upper limit a radius may change per layer. +1 to avoid rounding errors
+    const coord_t max_radius_change_per_layer = 1 + config.support_line_width / 2; // This is the upper limit a radius may change per layer. +1 to avoid rounding errors.
 
-    // smooth upwards
-    for (LayerIndex layer_idx = 0; layer_idx < LayerIndex(layer_tree_polygons.size()) - 1; layer_idx++)
+    // Smooth upward.
+    for (const auto layer_idx : ranges::views::iota(0UL, layer_tree_polygons.size() - 1UL))
```

Comment:
I do not think it is very relevant as i don't think that this code can even be reached in the relevant edge-case, but theoretically if `layer_tree_polygons.size() == 0` then this would underflow, while it would not underflow in my version as LayerIndex is a signed type.